infra/modules/stacks/compute_cluster/gke.tf

```diff
@@ -61,18 +61,63 @@ locals {
     }
   ]
 
+  # Spot node pools for cost-effective compute workloads
+  n2d_spot_node_pools = [for size in [8, 16, 32, 48, 64] : {
+    name           = "n2d-highmem-${size}-spot-nodes"
+    machine_type   = "n2d-highmem-${size}"
+    node_locations = local.default_node_locations
+    min_count      = 0
+    max_count      = 20 # Higher max count for spot instances
+    # NOTE: Local SSDs are only available to certain node groups, and hence cannot be set for the
+    # reguluar n2 nodes. The statement below allocates 2 SSDs to the node, each having a capacity of 375G.
+    # https://cloud.google.com/compute/docs/general-purpose-machines#n2d-high-mem
+    local_ssd_ephemeral_storage_count = size < 64 ? 2 : 4
```

Comment:
how are those SSDs actually utilizied? Don't they need to be mkfs'ed and mounted?

Reply:
fair enough